main/tests/test_string.h

```diff
@@ -31,12 +31,1009 @@
 #ifndef TEST_STRING_H
 #define TEST_STRING_H
 
+#include <thirdparty/doctest/doctest.h>
```

Comment:
`< >` is for system headers, so this should use `" "`.

I'd put it further down in its own section before the system includes.